server/game/player.js

```diff
@@ -305,12 +305,36 @@ class Player extends Spectator {
         this.challenges.setCannotInitiateForType(type, value);
     }
 
-    initDrawDeck() {
-        this.hand.each(card => {
+    resetCardPile(pile) {
+        pile.each(card => {
             card.moveTo('draw deck');
             this.drawDeck.push(card);
         });
+    }
+
+    resetDrawDeck() {
+        this.resetCardPile(this.hand);
         this.hand = _([]);
+
+        this.resetCardPile(this.cardsInPlay);
+        this.cardsInPlay = _([]);
+
+        this.resetCardPile(this.discardPile);
+        this.discardPile = _([]);
+
+        this.resetCardPile(this.deadPile);
+        this.deadPile = _([]);
+
+        _.each(this.additionalPiles, pile => {
+            this.resetCardPile(_(pile.cards));
+        });
+        this.additionalPiles = {};
```

Comment:
I'm pretty sure this will kill the Rains schemes plot.

Reply:
In my testing it doesn't appear to